km/km_vcpu_run.c

```diff
@@ -509,6 +510,9 @@ static void km_vcpu_one_kvm_run(km_vcpu_t* vcpu)
    }
 
    if (rc != 0) {
+      if (errno == 0) {
```

Comment:
I think the proper fix is to save errno in a local variable  immediately after ioctl( KVM_RUN ) returns.
And then when we discover ioctl() has failed, restore errno so that km_trace() will display it properly and the switch statement below this will be operating on a valid value.

Reply:
you right, will do